vocab/vocab.pyx

```diff
@@ -32,23 +32,68 @@ def alpha_tokenize(s):
     return [token for token in candidates if re_keep.search(token)]
 
 
-cdef class Stopwords:
-    def __cinit__(self):
-        # use a uint32_t for word ids
-
-        # Stop words may or may not be a part of the vocabulary (as unigrams),
-        # but we need to accumulate their counts for finding n-grams so they
-        # need an id.  So make them the largest uint32_t to avoid id conflicts
-        # with the non-vocab.
-        cdef uint32_t wordid = 4294967295
-        words = pkgutil.get_data('vocab', 'stop_words.txt').strip().split()
-        self.word2id.clear()
-        self.id2word.clear()
-        for word in words:
-            self.min_id = wordid
-            self.word2id[word] = wordid
-            self.id2word[wordid] = word
-            wordid -= 1
+cdef load_stopwords(set[string]& stopwords):
+    words = pkgutil.get_data('vocab', 'stop_words.txt').strip().split()
+    stopwords.clear()
+    for word in words:
+        stopwords.insert(word)
+
+
+cdef set[string] STOP_WORDS
+load_stopwords(STOP_WORDS)
+
+
+class IndexLookupTable(object):
+    def __init__(self, counts, table_size, power):
+        self._table_size = table_size
+        self._power = power
+        self._table = IndexLookupTable._build(counts, table_size, power) \
+            if counts else None
+
+    def update_table(self, counts):
+        """ Update the table with new counts"""
+        # just build a new table with the counts
+        self._table = IndexLookupTable._build(counts,
+                                              self._table_size, self._power)
+
+    @staticmethod
+    def _build(counts, table_size, power):
+        """
+        Create a table using the vocabulary tokens counts, which can be used
+        for drawing random words in negative sampling routines (useful for
+        word2gauss.)
+        This function is called by the constructor if the build_table
```

Comment:
To simplify the interface, I'd removed the option, but forgot to update the comment.